### PR TITLE
fix: added url box to public event urls

### LINKED ISSUE #6331: Specific Event Dashboard: Add URL Box
The event dashboard shows a URL to the public page of the event. Please make change this to a box similar to the video URL box on the video tab.

Screenshot from 2021-01-12 10-19-22

Screenshot from 2021-01-12 10-01-40

### PR DESCRIPTION
**Pull request by Creatoon**
Fixes #6331 

#### Short description of what this resolves:
This PR adds url box over public url's

If the event is live
Screenshot_2021-01-12 js engines Events Open Event

if the event is not live
Screenshot_2021-01-12 Jio WIfi Events Open Event

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
1. Please keep the font size and position as before.
2. Could you also change the following as part of this issue, please:
The column width of the top box table is different to the column width in the sessions and speakers box. Please make the column width in the overview box the same as in the sessions and speakers box.

Screenshot from 2021-01-12 18-52-04

**Comment by Creatoon** [comment]
>     1. Please keep the font size and position as before.
> 
>     2. Could you also change the following as part of this issue, please:
>        The column width of the top box table is different to the column width in the sessions and speakers box. Please make the column width in the overview box the same as in the sessions and speakers box.
> 
> 
> Screenshot from 2021-01-12 18-52-04

ok sir working

**Comment by Creatoon** [comment]
@mariobehling After changes 

- [ ] Please keep the font size and position as before.
- [ ] The column width of the top box table is different to the column width in the sessions and speakers box. Please make the column width in the overview box the same as in the sessions and speakers box.
 
Screenshot :-

Previous - 

Screenshot_2021-01-13 Company information Events eventyay

Now, with both the task you mentioned

Screenshot_2021-01-13 js engines Events Open Event

please mark if i am going wrong, Thanks.

**Comment by Creatoon** [comment]
@mariobehling @iamareebjamal sir please review

File: app/templates/components/url-box.hbs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <div class="ui action input" style="width: 100%; height: 36px">
-  <input class="truncate" style="flex: 1 0 !important" type="text" value={{@url}} placeholder={{@url}} readonly>
+  <input class="truncate" style={{html-safe (concat "flex: 1 0 !important;" @extraStyles)}} type="text" value={{@url}} placeholder={{@url}} readonly>
```
Comment (iamareebjamal) [inline app/templates/components/url-box.hbs:2]: why

**Comment by iamareebjamal** [comment]
Too many changes

**Comment by mariobehling** [comment]
Not working as expected 

Screenshot from 2021-01-13 20-50-46

File: app/templates/components/url-box.hbs (hunk 1 above)
Comment (Creatoon) [reply app/templates/components/url-box.hbs:2]:
```
<input class="truncate" style={{"flex: 1 0 !important;" @extraStyles)}} type="text" value={{@url}} placeholder={{@url}}>
```

Sir this ember thrown this error on runtime , when i was adding some more styles to the url-box.

```
Binding style attributes may introduce cross-site scripting vulnerabilities; please ensure that values being bound are properly escaped. 
```
Then i used concat to add styles

```
Concatenated styles must be marked as `htmlSafe`.  style-concatenation
```
that's why i used html safe keyword

if i will not use extra stylings then, outcome will be like this

Screenshot_2021-01-14 js engines Events Open Event

but mario sir asked me to keep the positions and everything same.

**Comment by Creatoon** [comment]
> Not working as expected
> 

 sir will this work?
Screenshot_2021-01-14 js engines Events Open Event(2)

**Comment by Creatoon** [comment]
> Not working as expected
> 
 @mariobehling will this work? i will commit changs after your confirmation.

https://user-images.githubusercontent.com/65965202/104539014-d9fd0480-5642-11eb-8f6d-0cec746392c3.mp4

**Comment by Creatoon** [comment]
> Too many changes

fixed

**Comment by Creatoon** [comment]
@mariobehling sir please look the ui, will this work? ,thanks

**Comment by Creatoon** [comment]
@mariobehling

 sir on big screens,
Screenshot_2021-01-16 Community meet Events Open Event(1)

On mobile view
Screenshot_2021-01-16 Community meet Events Open Event

also fixed the column width you mentioned previously , 
if this do not works just let me know, i will improve this, thanks

**Comment by Creatoon** [comment]
@iamareebjamal sir please look the changes

**Review by mariobehling — CHANGES_REQUESTED** [review]
Not working as expected if text is longer in some languages.

Screenshot from 2021-01-16 23-07-44

**Comment by mariobehling** [comment]
The font inside the box does not need to be a heading. Just use standard size here, please. Thanks

**Comment by mariobehling** [comment]
Height of the area has also increase, but we do not need the area to be even more prominent than before. Previous height is sufficient.

Screenshot from 2021-01-16 23-11-58

**Comment by Creatoon** [comment]
@mariobehling @iamareebjamal Sir , previously there was not url box , there was only a text "your event is live at" and an anchor tag containing url, they both were inline elements that's why they get's easily break into multiple lines, but in this situation.

url box and the text "your event is live at" are both block level elemnts they have width 100%, we can either put both "your event is live at" and url box in same line or in two different lines, but if you want them to be in same line and also expect that they both get stacked into different columns for a specific "TEXT LENGTH (your event is live at => in different languages)" then this won't be possible only with css,  because we can only add media queries for different widths not for different text lengths.

as i mentioned above , if we put both in single line , then for small screen size
Screenshot_2021-01-17 hero motocorp Events Open Event

or if we put them in different lines then,
Screenshot_2021-01-17 hero motocorp Events Open Event(1)

from my side 2nd option is better because for long text ("your event is live at => in different lang") first option will break the ui.

I have tried many things from flex to grids and both will not work here , we have to chose one.

**Pull request closed without merging**